specification/cognitiveservices/data-plane/Language/preview/2022-02-01-preview/textanalytics.json

```diff
@@ -0,0 +1,3029 @@
+          }
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/PreBuiltResult"
+        }
+      ],
+      "required": [
+        "documents"
+      ]
+    },
+    "LanguageDetectionDocumentResult": {
+      "type": "object",
+      "properties": {
+        "detectedLanguage": {
+          "description": "Detected Language.",
+          "$ref": "#/definitions/DetectedLanguage"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/DocumentResult"
+        }
+      ],
+      "required": [
+        "detectedLanguage"
+      ]
+    },
+    "DetectedLanguage": {
+      "type": "object",
+      "required": [
+        "name",
+        "iso6391Name",
+        "confidenceScore"
+      ],
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Long name of a detected language (e.g. English, French)."
+        },
+        "iso6391Name": {
+          "type": "string",
+          "description": "A two letter representation of the detected language according to the ISO 639-1 standard (e.g. en, fr)."
+        },
+        "confidenceScore": {
+          "type": "number",
+          "format": "double",
+          "description": "A confidence score between 0 and 1. Scores close to 1 indicate 100% certainty that the identified language is true."
+        }
+      }
+    },
+    "AnalyzeTextJobState": {
+      "allOf": [
+        {
+          "$ref": "#/definitions/JobState"
+        },
+        {
+          "$ref": "#/definitions/TasksState"
+        },
+        {
+          "$ref": "#/definitions/AnalyzeTextJobStatistics"
+        }
+      ]
+    },
+    "Pagination": {
+      "properties": {
+        "nextLink": {
+          "type": "string"
+        }
+      },
+      "type": "object"
+    },
+    "JobMetadata": {
+      "properties": {
+        "displayName": {
+          "type": "string"
+        },
+        "createdDateTime": {
+          "format": "date-time",
+          "type": "string"
+        },
+        "expirationDateTime": {
+          "format": "date-time",
+          "type": "string"
+        },
+        "jobId": {
+          "format": "uuid",
+          "type": "string"
+        },
+        "lastUpdateDateTime": {
+          "format": "date-time",
+          "type": "string"
+        },
+        "status": {
+          "enum": [
+            "notStarted",
+            "running",
+            "succeeded",
+            "partiallySucceeded",
+            "failed",
+            "cancelled",
+            "cancelling"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "modelAsString": false,
+            "name": "State"
+          }
+        }
+      },
+      "required": [
+        "jobId",
+        "lastUpdateDateTime",
+        "createdDateTime",
+        "status"
+      ],
+      "type": "object"
+    },
+    "JobState": {
+      "allOf": [
+        {
+          "$ref": "#/definitions/JobMetadata"
```

Comment:
I would consider folding in JobMetadata, JobErrors, Pagination into the JobState model. Is there a specific reason that we want to keep them separate (future extensibility)?

Reply:
I think this consolidation makes sense, since the new LROs we have upcoming will follow this same structure and it would be fairly easy to separate again later if necessary.  I will add this shortly.

Reply:
Done!